kernel/aster-nix/src/thread/task.rs

```diff
@@ -36,15 +39,22 @@ pub fn create_new_user_task(user_space: Arc<UserSpace>, thread_ref: Weak<Thread>
             user_mode.context().syscall_ret()
         );
 
-        let posix_thread = current_thread.as_posix_thread().unwrap();
-        let has_kernel_event_fn = || posix_thread.has_pending();
+        let has_kernel_event_fn = || current_posix_thread.has_pending();
         loop {
             let return_reason = user_mode.execute(has_kernel_event_fn);
             let context = user_mode.context_mut();
             // handle user event:
             match return_reason {
-                ReturnReason::UserException => handle_exception(context),
-                ReturnReason::UserSyscall => handle_syscall(context),
+                ReturnReason::UserException => handle_exception(&current_process, context),
+                ReturnReason::UserSyscall => handle_syscall(
+                    CurrentInfo {
+                        process: current_process.as_ref(),
+                        posix_thread: current_posix_thread,
+                        thread: current_thread.as_ref(),
+                        task: current_task.as_ref(),
+                    },
+                    context,
+                ),
```

Comment:
How do you like passing `&CurrentInfo` instead of `CurrentInfo`, then `CurrentInfo` can be built before the loop?

The benefit is that `&CurrentInfo` is 8 bytes while `CurrentInfo` is 32bytes, the cost is that there may be another indirect access when visiting `CurrentInfo` (through reference).

Reply:
There would be no difference in release mode though.